lib/src/space/grounding.rs

```diff
@@ -56,6 +77,14 @@ fn atom_to_trie_key(atom: &Atom) -> TrieKey<SymbolAtom> {
                 expr.children().iter().for_each(|child| fill_key(child, tokens));
                 tokens.push(TrieToken::RightPar);
             },
+            // FIXME, see below
```

Comment:
@vsbogd to comment.  But I think this change will mean that custom grounded matchers will not run if the type successfully serializes and the hashes aren't equal.  Which I think is against the design of HE as it currently stands.

It probably a non-issue right now, in practice, since the only types that currently support serialization are numbers and bools, and there is only one implementation of those.  But I think this cuts against the intentions of the design.

Reply:
Change is quite big https://github.com/Adam-Vandervorst/hyperon-experimental/compare/main...vsbogd:hyperon-experimental:grounded-matching?expand=1 so I think I will merge it and will make additional changes as separate PRs.